src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -222,7 +222,8 @@ static uint8_t pin_num_to_exti_irq(uint16_t pin_num)
 
 static term create_pair(Context *ctx, term term1, term term2)
 {
-    term ret = term_alloc_tuple(2, &ctx->heap);
+    BEGIN_WITH_STACK_HEAP(TUPLE_SIZE(2), heap);
```

Comment:
I need to double check, however this might cause a VM crash.

Reply:
Or `RAISE_ERROR(BADARG_ATOM)` in C, for the equivalent behavior as `error(badarg)` in Erlang.  (`throw/1`, `error/1`, and `exit/1` all have different semantics in Erlang, and in some cases, different behavior)